scapy/config.py

```diff
@@ -845,6 +845,7 @@ class Conf(ConfClass):
         'eap',
         'gprs',
         'hsrp',
+        'ikev2',
```

Comment:
In that case shouldn't we let it remain in `contribs/` ?

Reply:
I am fine with moving it to layers as long as we think that the layer is good quality.

Reply:
@guedou what is [usually found on common networks](https://github.com/secdev/scapy/blob/master/doc/scapy/build_dissect.rst#add-new-protocols-to-scapy) also depends on the personal point-of-view of the observer. However, I would never dare to challenge you on this question :wink:.

Instead, I'd rather play a different card and argue for consistency:  Whether ISAKMP is a common protocol or not might be debatable, but the decision was made long time ago: ISAKMP is an official layer. Now from my viewpoint, ISAKMP is essentially IKEv1, and IKEv2 is its version upgrade. The standardization bodies fixed a lot of mistakes and deficiencies of IKEv1, so IKEv2 is a *far superior* than version one, and it will be only a matter of time until it supersedes it.  This might also be the reason, why IKEv1 support couldn't be added with a few ifs-and-elses to the ISAKMP class. But de facto, it is a version upgrade.

IMHO  the true reason why IKEv2 never left the contrib status is because it was not mature enough. Loading scapy.contrib.ikev2 broke the IKEv1 bindings. When I started to fix this, I initially did not intend to move ikev2.py. But all attempts to retrofit IKEv2 on load turned out to be ugly somehow. The current solution, having a dispatch hook in ISAKMP which checks the IKE version is the most natural choice. But this can only be done if IKEv2 is in layers, not contrib.

My pull request fixes not only coexistence of IKEv1 and IKEv2, but also the missing bindings for NAT-traversal, giving the user a smooth experience.  Don't you agree that with my changes IKEv2 is now mature enough to deserve a promotion to an official Scapy layer?  

> I am fine with moving it to layers as long as we think that the layer is good quality.

Reading your last comment, my hope returned that I can still convince you with good quality. And as you can see from the PR list, this one is not the end of the story, there are more PRs in the queue.

Reply:
> @mspncp we can keep the commits history if you ask for it, otherwise we squash everything.

Yes, please!